alf/environments/suite_safety_gym.py

```diff
@@ -0,0 +1,201 @@
+1. All have distance-based dense rewards
+2. All have continual goals: after reaching a goal, the goal is reset but the
+   layout keeps the same until timeout.
+3. Layouts are randomized before episodes begin
+4. Costs are indicator binaries (0 or 1). Every positive cost will be binarized
+   to 1. Thus the total cost will be 1 if any component cost is positive.
+5. level 0 has no constraints; level 1 has some unsafe elements; level 2 has
+   very dense unsafe elements.
+
+See https://github.com/openai/safety-gym/blob/f31042f2f9ee61b9034dd6a416955972911544f5/safety_gym/envs/engine.py#L97
+for a complete list of default configurations.
+"""
+
+try:
+    import mujoco_py
+    import safety_gym
+except ImportError:
+    mujoco_py = None
+    safety_gym = None
+
+import functools
+import numpy as np
+import gym
+
+import gin
+from alf.environments import suite_gym, alf_wrappers, process_environment
+
+
+def is_available():
+    return (mujoco_py is not None and safety_gym is not None)
+
+
+class CompleteEnvInfo(gym.Wrapper):
+    """Always set the complete set of information so that the env info has a
+    fixed shape (no matter whether some event occurs or not), which is required
+    by ALF.
+
+    The current safety gym env only adds a key to env info when the corresponding
+    event is triggered, see:
+    https://github.com/openai/safety-gym/blob/f31042f2f9ee61b9034dd6a416955972911544f5/safety_gym/envs/engine.py#L1242
+    """
+
+    def __init__(self, env, env_name):
+        super().__init__(env)
+        # env info keys are retrieved from:
+        # https://github.com/openai/safety-gym/blob/master/safety_gym/envs/engine.py
+        self._env_info_keys = [
+            'cost_exception',
+            'goal_met',
+            'cost'  # this is the summed overall cost
+        ]
+        if not self._is_level0_env(env_name):
+            # for level 1 and 2 envs, there are constraints cost info
+            self._env_info_keys += [
+                'cost_vases_contact', 'cost_pillars', 'cost_buttons',
+                'cost_gremlins', 'cost_vases_displace', 'cost_vases_velocity',
+                'cost_hazards'
+            ]
+
+    def _is_level0_env(self, env_name):
+        return "0-" in env_name
+
+    def _generate_default_env_info(self):
+        env_info = {}
+        for key in self._env_info_keys:
+            if key == "goal_met":
+                env_info[key] = False
+            else:
+                env_info[key] = np.float32(0.)
+        return env_info
+
+    def step(self, action):
+        env_info = self._generate_default_env_info()
+        obs, reward, done, info = self.env.step(action)
+        env_info.update(info)
+        return obs, reward, done, env_info
+
+
+class VectorReward(gym.Wrapper):
+    """This wrapper makes the env returns a reward vector of length 3. The three
+    dimensions are:
+
+    1. distance-improvement reward indicating the delta smaller distance to the goal
```

Comment:
The original reward from safety gym seems more complicated than what you said here (https://github.com/openai/safety-gym/blob/f31042f2f9ee61b9034dd6a416955972911544f5/safety_gym/envs/engine.py#L1306)

Reply:
They have lots of options for rewards. Some for debugging purposes or customized environments. But in the public configured environments, only the distance delta (first three options) is used
https://github.com/openai/safety-gym/blob/f31042f2f9ee61b9034dd6a416955972911544f5/safety_gym/envs/suite.py#L90